### PR TITLE
decomp some effect funcs in hos_05

### PR DESCRIPTION
**Pull request by Wrymouth**

### CODE REVIEW

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 1]
```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
 // adjusts properties of EmitterVolume:GoldShimmer2 effect
-INCLUDE_ASM(s32, "world/area_hos/hos_05/A2AAC0", func_80240E50_A2B090);
+ApiStatus func_80240E50_A2B090(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 posX;
+    s32 posY;
+    f32 posZ;
+    EffectInstance* effect;
+
+    effect = (EffectInstance*) evt_get_variable(script, *args++);
```
Comment (ethteck) [inline src/world/area_hos/hos_05/A2AAC0.c:208]: Could you move this under the args initialization and initialize it on the same line you declare it?

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 2]
```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
+    evt_get_variable(script, *args++);
+    posX = evt_get_float_variable(script, *args++);
+    posY = evt_get_float_variable(script, *args++);
+    posZ = evt_get_float_variable(script, *args++);
+    effect->data.miscParticles->pos.x = (f32) posX;
```
Comment (ethteck) [inline src/world/area_hos/hos_05/A2AAC0.c:213]: no casts needed (including the double cast on the last one)

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 3]
```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
+    effect->data.miscParticles->pos.z = (f32) (s32) posZ;
+    return ApiStatus_DONE2;
+}
+
+
+ApiStatus func_80240F30_A2B170(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    EffectInstance* effect = (EffectInstance*) evt_get_variable(script, ArrayVar(0));
+    effect->data.somethingRotating[D_802495DC_A3381C + 1].unk_29 = 1;
+    D_802495DC_A3381C += 1;
```
Comment (ethteck) [inline src/world/area_hos/hos_05/A2AAC0.c:224]: ++

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 4]
```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
+}
 
-INCLUDE_ASM(s32, "world/area_hos/hos_05/A2AAC0", func_80240F30_A2B170);
 
-INCLUDE_ASM(s32, "world/area_hos/hos_05/A2AAC0", func_80240F88_A2B1C8);
+ApiStatus func_80240F88_A2B1C8(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    EffectInstance* effect = (EffectInstance*) evt_get_variable(script, ArrayVar(0));
+    effect->data.somethingRotating[D_802495E0_A33820 + 1].unk_29 = 3;
+    D_802495E0_A33820 += 1;
```
Comment (ethteck) [inline src/world/area_hos/hos_05/A2AAC0.c:233]: ++

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 5]
```diff
@@ -196,13 +198,52 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240FE0_A2B220(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    EffectInstance* effect;
```
Comment (ethteck) [inline src/world/area_hos/hos_05/A2AAC0.c:239]: init on the same line you declare it, one empty line after var decls pls

**Review by ethteck — CHANGES_REQUESTED** [review]
nice yorb :) some small changes

File: src/world/area_hos/hos_05/A2AAC0.c (hunk 2 above)
Comment (pixel-stuck) [reply src/world/area_hos/hos_05/A2AAC0.c:213]: also probably can move these inline with the var declarations as well (?) unless we don't want to include the useless var

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀 This PR matches 4 functions (+0.01%) on `us`.

**Review by ethteck — APPROVED** [review]

File: src/world/area_hos/hos_05/A2AAC0.c [hunk 6]
```diff
@@ -196,13 +198,50 @@ ApiStatus func_80240E30_A2B070(Evt* script, s32 isInitialCall) {
...
+    s32 posX = evt_get_float_variable(script, *args++);
+    s32 posY = evt_get_float_variable(script, *args++);
+    s32 posZ = evt_get_float_variable(script, *args++);
+
+    effect->data.miscParticles->pos.x = posX;
+    effect->data.miscParticles->pos.y = posY;
+    effect->data.miscParticles->pos.z = posZ;
+    return ApiStatus_DONE2;
+}
+
```
Comment (nanaian) [inline src/world/area_hos/hos_05/A2AAC0.c:214]: nit: use 1 empty line between blocks

**Review by nanaian — CHANGES_REQUESTED** [review]
Good matches :)

File: src/world/area_hos/hos_05/A2AAC0.c (hunk 6 above)
Comment (Wrymouth) [reply src/world/area_hos/hos_05/A2AAC0.c:214]: An empty line before the return, you mean? Should I do that for every func then?

**Review by nanaian — APPROVED** [review]

**Pull request merged**